### PR TITLE
[Swift] Strip spare bits from pointers before printing them.

### PR DESCRIPTION
**Pull request by dcci**
SR-7565
<rdar://problem/39869332>

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci please test

**Comment by dcci** [comment]
@swift-ci please test

File: packages/Python/lldbsuite/test/lang/swift/enum_tagged/main.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    init(z: Void) {
+        variant = .z
+    }
+}
+
+let x = A(x: 42)
+let y = A(y: 39)
+print("!")  //%self.expect("frame var -d run-target -- x", substrs=['(xx = 42)'])
```
Comment (adrian-prantl) [inline packages/Python/lldbsuite/test/lang/swift/enum_tagged/main.swift:40]: I recently learned about `lldbutil.check_variable()`, which seems to be slightly less fragile / string-based than self.expect. Is it possible to use it inside of an inline test?

File: source/Target/SwiftLanguageRuntime.cpp [hunk 2]
```diff
@@ -2639,6 +2639,26 @@ bool SwiftLanguageRuntime::FixupReference(lldb::addr_t &addr,
     }
   }
   default:
+    // Adjust the pointer to strip away the spare bits.
```
Comment (adrian-prantl) [inline source/Target/SwiftLanguageRuntime.cpp:2642]: Can you add a comment whether this is runtime or compiler-specific?

File: source/Target/SwiftLanguageRuntime.cpp [hunk 3]
```diff
@@ -2639,6 +2639,26 @@ bool SwiftLanguageRuntime::FixupReference(lldb::addr_t &addr,
...
+      addr &= ~SWIFT_ABI_ARM_SWIFT_SPARE_BITS_MASK;
+      break;
+    case llvm::Triple::ArchType::x86:
+      addr &= ~SWIFT_ABI_I386_SWIFT_SPARE_BITS_MASK;
+      break;
+    case llvm::Triple::ArchType::x86_64:
+      addr &= ~SWIFT_ABI_X86_64_SWIFT_SPARE_BITS_MASK;
+      break;
+    default:
+      llvm_unreachable("unsupported arch");
```
Comment (adrian-prantl) [inline source/Target/SwiftLanguageRuntime.cpp:2659]: This is not unreachable. Swift supports powerpc and whatever arch z/os uses.

Comment (adrian-prantl) [reply source/Target/SwiftLanguageRuntime.cpp:2659]: I would just leave the value as is on other arches.

Comment (dcci) [reply source/Target/SwiftLanguageRuntime.cpp:2659]: OK.

File: packages/Python/lldbsuite/test/lang/swift/enum_tagged/main.swift (hunk 1 above)
Comment (jimingham) [reply packages/Python/lldbsuite/test/lang/swift/enum_tagged/main.swift:40]: The commands in lldbinline.py are run by "exec(command_string") in a method of the test class defined in lldbinline.py.  That file imports lldbutil.  IIUC how python exec works, the command string should see whatever the surrounding context sees.  So you should be able to use anything from lldbutil in the  test commands.

**Pull request merged**